source/common/http/conn_manager_impl.cc

```diff
@@ -180,10 +180,12 @@ void ConnectionManagerImpl::doEndStream(ActiveStream& stream) {
   // explicitly nulls out response_encoder to avoid the downstream being notified of the
   // Envoy-internal stream instance being ended.
   if (stream.response_encoder_ != nullptr &&
-      (!stream.state_.remote_complete_ || !stream.state_.local_complete_)) {
```

Comment:
Are we missing a local_complete_to_codec_ in ::responseDataTooLarge?
That's the one special case where we do response_encoder->encodeHeaders/encodeData that's not using ::sendLocalReply (because we're intentionally bypassing the filter chain)

unclear if we could hit resetAllStreams under that stack but I think we should handle it for completeness.

Reply:
We call `maybeEndEncode(...)` in that path so I think we are ok?

Reply:
Oh yeah, and local_complete_ will be true both on gRPC and not-gRPC path.  Cool.